frontend/src/js/components/Store.tsx

```diff
@@ -125,10 +125,42 @@ export default class Store extends React.Component<IStoreProps, IStoreState> {
           reservationError: null
         };
         if (hasConflictingSeats) {
+          // If we don't have numbered seats, we don't actually care about the seat id's. To keep the backend simple,
+          // we have pseudo-seats still, so in frontend we have to choose some id's. Now, if those seats happen to be
+          // taken by some concurrent buyer, our 'reserveTickets'-call will fail, but we don't need any user action.
+          // So let's just get some new free id's and retry. This will probably only happen like 0 times ever, so there
+          // is no need to worry about performance. If there isn't enough free seats left anymore, we show the error.
+          // Note that this supports multiple sections, making the code a bit more complicated.
+          if(this.venue.ticket_type === 'generic-tickets') {
+            var enoughTicketsLeft = true;
+            var conflictingTickets = _.filter(this.tickets, (t: Ticket) => _.contains(conflictingSeatIds, t.get('seat').id));
+            chosenSeatIds = this.tickets.map((t: Ticket): number => t.get('seat').id);
+            for(var i = 0; i < conflictingTickets.length; i++) {
```

Comment:
btw, there is a super-cool `for(var a of array)` statement in Typescript (https://basarat.gitbooks.io/typescript/content/docs/for...of.html)